### ISSUE TITLE
Class members (variables/functions) show up in the documentation twice (Origin: bugzilla #155284)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#1300]
status NEW severity *minor* in component *build* for *---*
Reported in version *1.5.3-SVN* on platform *Other*
Assigned to: Dimitri van Heesch

On 2004-10-13 11:01:53 +0000, Rotaru Calin wrote:

    I document my code in the following way (keeps my code clean and easy to
    read):
    
    *.cpp - source file having only the header for the file and brief comments
    for
    the global variables declared inside the file
    *.h - header file having as for .cpp only header and variables as doxygen
    documentation
    *.doxygen - specific doxygen comment file in which all
    functions/classes/variables from the corresponding .cpp and .h files are
    documented.
    
    Doxygen in version 1.3.8 had no problem with this layout (I am not 100%
    sure,
    but I do not remember seeing it). 1.3.9 produces (funny thing is that not
    always
    -i will document below-) in HTML output two declarations insted of one. The
    first one is documented and links right to its doecumentation, the second
    one is
    not documented and therefore no link at all. The declarations are duplicated
    blockwise (that is first all members appear right, then all appear without
    links).
    
    I thought originally that the problem was related to the inclusion process.
    I am
    not sure anymore. I have a base class and four derived classes, all five
    files
    documented as described at the beginning (in .doxygen files). The base class
    shows duplicates, from the derived classes one has duplicated members, three
    are
    ok. All are basically the same (created with copy/paste) and the .h are
    included
    the same number of times in the project (once in the .cpp file, once in the
    base
    .cpp file).
    
    Hope that this helps identifying the bug. If not drop me a line, I may
    probably
    be able to generate a simple project layout to exemplify.
    
    Calin

On 2004-10-13 11:22:59 +0000, Rotaru Calin wrote:

    Additional info:
    
    The bug does not apply to member variables (or I don't see it anymore).
    The bug does not apply to implicit inline (inside the class defined)
    functions.
    
    i.e. The bug applies to normal (out of the class defined functions) no
    matter if
    they are virtual or not, inherited or not, abstract or not.
    
    Calin

On 2004-10-13 11:55:10 +0000, Rotaru Calin wrote:

    Bug located:
    
    The bug appears when a member variable is documented twice (I had some old
    copies of some of the files in a subdirectory). The bug should read now:
    ---
    
    No warning is generated if commenting twice a class function. Of course with
    lower priority and importance.
    
    Calin

On 2004-10-13 20:00:51 +0000, Dimitri van Heesch wrote:

    This would be better indeed, but this is difficult to implement because in
    some
    situations were a function is documented only once, it's documentation is
    still
    added twice, making it difficult to suppress a warning in such case while
    generating one if a function is really documented more than once.

On 2007-10-15 19:40:39 +0000, Dimitri van Heesch wrote:

    This bug's version was set to "latest". Since this is a moving target, I
    changed it to 1.5.3-SVN. If you believe this has already been fixed, then
    please change the status accordingly.
